cli/src/formatted_diagnostic.rs

```diff
@@ -0,0 +1,156 @@
+use std::collections::HashMap;
+use std::fmt::{Debug, Display};
+use std::io;
+use std::ops::Add;
+
+use analyzer::diagnostic::ObservationTag;
+use analyzer::relations::SourceId;
+use ariadne::{Cache, Color, ColorGenerator, Config, Label, Report, ReportKind};
+use regex::Regex;
+use yansi::Paint;
+
+use context::source::Source;
+
+use crate::fix_ariadne_report;
+
+pub fn render_diagnostic<'a>(
+    diagnostic: analyzer::diagnostic::Diagnostic,
+    source_supplier: impl Fn(SourceId) -> Source<'a>,
+) -> io::Result<String> {
+    let mut colors = ColorGenerator::new();
+    let mut colormap = HashMap::new();
```

Comment:
colormap is never inserted, and is only used for `ObservationTag::Other` that shouldn't have a magic color value.

Reply:
yes that was the goal of ObservationTag::Other, group observations by "other" tags and then the cli would have a different color for those groups